packages/boxel-ui/addon/src/components/resizable-panel-group/index.gts

```diff
@@ -365,109 +408,50 @@ export default class ResizablePanelGroup extends Component<Signature> {
+    );
+    if (remainingContainerSize > 0) {
+      panelLengths[0] = panelLengths[0]! + remainingContainerSize;
+    } else if (remainingContainerSize < 0 && panelWithNoMinLength) {
+      panelLengths[panelWithNoMinLength[0] - 1] =
+        panelLengths[panelWithNoMinLength[0] - 1]! + remainingContainerSize > 0
+          ? panelLengths[panelWithNoMinLength[0] - 1]! + remainingContainerSize
+          : 0;
     }
 
-    if (totalPanelLength > panelGroupWithoutResizeHandlerLength) {
-      // length decreases needed
-      let remainingDecreaseNeeded = panelGroupRemainingLength * -1;
-      while (remainingDecreaseNeeded > 0) {
-        for (let index = 1; index <= this.listPanelContext.size; index++) {
-          let panelContext = this.listPanelContext.get(index);
-          let panelLength: number = panelLengths[index - 1] || 0;
-          let proportionalDecrease =
-            (panelLength / sumArray(panelLengths)) * remainingDecreaseNeeded;
-          let actualDecrease = Math.ceil(
-            panelContext?.minLengthPx
-              ? Math.min(proportionalDecrease, panelContext.minLengthPx)
-              : proportionalDecrease,
-          );
-          remainingDecreaseNeeded = remainingDecreaseNeeded - actualDecrease;
-          panelLengths[index - 1] = panelLength - actualDecrease;
-        }
-      }
-      for (let index = 1; index <= this.listPanelContext.size; index++) {
-        let panelContext = this.listPanelContext.get(index);
-        if (panelContext) {
-          this.listPanelContext.set(index, {
-            ...panelContext,
-            lengthPx: panelLengths[index - 1] || 0,
-          });
-        }
-      }
-    } else {
-      // length increases needed
-      let largestPanel = {
-        id: 1,
-        length: 0,
-      };
-      for (let index = 1; index <= this.listPanelContext.size; index++) {
-        let panelLength = panelLengths[index - 1] || 0;
-        let newLength =
-          panelLength +
-          Math.round(
-            (panelLength / totalPanelLength) * panelGroupRemainingLength,
-          );
-        let panelContext = this.listPanelContext.get(index);
-        if (panelContext) {
-          this.listPanelContext.set(index, {
-            ...panelContext,
-            lengthPx: newLength,
-          });
-        }
-        if (largestPanel.length < newLength) {
-          largestPanel.id = index;
-          largestPanel.length = newLength;
-        }
-
-        panelGroupWithoutResizeHandlerLength =
-          panelGroupWithoutResizeHandlerLength - newLength;
-      }
-
-      let largestPanelContext = this.listPanelContext.get(largestPanel.id);
-      if (panelGroupWithoutResizeHandlerLength > 0 && largestPanelContext) {
-        this.listPanelContext.set(largestPanel.id, {
-          ...largestPanelContext,
-          lengthPx: largestPanel.length + panelGroupWithoutResizeHandlerLength,
+    for (let index = 1; index <= this.listPanelContext.size; index++) {
```

Comment:
consider either a) using a zero based position system, or b) naming this "position" instead of "index"

Reply:
lean toward zero-based for everything -- it's what developers expect